### PR TITLE
Load theme translations files automatically

### LINKED ISSUE #1077: Load theme translations files automatically
In order to allow theme developers to add their own translations, that would be great to have a way to put translation files somewhere in the theme folder and to make Shaarli load them automatically.

### PR DESCRIPTION
**Pull request by ArthurHoaro**
Fixes #1077

Take a look at the docs update to see how it works

### CODE REVIEW

File: tests/utils/customtpl/dummy/language/fr/LC_MESSAGES/dummy.po [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+"Last-Translator: \n"
+"Language-Team: Shaarli\n"
+"Language: fr_FR\n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: 8bit\n"
+"Plural-Forms: nplurals=2; plural=(n > 1);\n"
+"X-Generator: Poedit 2.0.6\n"
+
+msgid "cock"
```
Comment (virtualtam) [inline tests/utils/customtpl/dummy/language/fr/LC_MESSAGES/dummy.po]: this test should use reference data that is not likely to be interpreted as being suggestive :)

File: application/Languages.php [hunk 2]
```diff
@@ -122,6 +128,19 @@ protected function initPhpTranslator()
             $this->translator->loadTranslations($translations);
         } catch (\InvalidArgumentException $e) {}
 
+        // Default extension translation from the current theme
+        $theme = $this->conf->get('theme');
+        $themeTransFolder = rtrim($this->conf->get('raintpl_tpl'), '/') .'/'. $theme .'/language';
+        if (is_dir($themeTransFolder)) {
+            try {
+                /** @var Translations $translations */
```
Comment (virtualtam) [inline application/Languages.php]: [translation? translation translation. translation!](https://www.youtube.com/watch?v=Q6Fuxkinhug)

Comment (ArthurHoaro) [reply application/Languages.php]: I use this for type hinting in the IDE but the lib should handle it.

File: tests/utils/customtpl/dummy/language/fr/LC_MESSAGES/dummy.po (hunk 1 above)
Comment (ArthurHoaro) [reply tests/utils/customtpl/dummy/language/fr/LC_MESSAGES/dummy.po]: WHAT'S WRONG WITH OUR PROUD NATIONAL SYMBOL?!

**Comment by ArthurHoaro** [comment]
Rebased.

File: tests/utils/customtpl/dummy/language/fr/LC_MESSAGES/dummy.po (hunk 1 above)
Comment (virtualtam) [reply tests/utils/customtpl/dummy/language/fr/LC_MESSAGES/dummy.po]: Seems alright to me ;-)

**Pull request merged**